core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -398,164 +410,185 @@ public SyncStatus sync() throws AccessControlException, InvalidPathException {
       return SyncStatus.NOT_NEEDED;
     }
     Instant startTime = Instant.now();
-    try (LockedInodePath path =
-             mInodeTree.lockInodePath(mRootScheme, mRpcContext.getJournalContext())) {
-      if (mAuditContext != null && mAuditContextSrcInodeFunc != null) {
-        mAuditContext.setSrcInode(mAuditContextSrcInodeFunc.apply(path));
-      }
+    try (LockedInodePath path = lockSyncRoot()) {
```

Comment:
previously, sync() method syncs the root first, with this LockedInodePath. Then the lock is released and we start many threads to sync its children and these threads will acquire corresponding locks for the paths.

In this implementation, this lock will be acquired here and release after the whole sync process finishes now. But once the root sync finishes, it will be downgraded to the read mode so that other requests can access the directory.

Add an inner try for the original try...with resource.